### ISSUE TITLE
False positive for `derive_hash_xor_eq`

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#4658]
`derive_hash_xor_eq` is triggered for custom traits that are named `Hash`.

**Bad**
```rust
#[derive(PartialEq)]
struct Foo;

trait Hash {}

impl Hash for Foo {}
```

**Ok**
```rust
#[derive(PartialEq)]
struct Foo;

trait Hash {}

impl self::Hash for Foo {}
```
